### PR TITLE
Start of Autotoolification.

### PR DESCRIPTION
**Pull request by ams**
This just gets the basics working on GNU/Linux.  It does not try hard
to figure out things.

I consider this to be a work-in-progress for now.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-dns])])],
+    [enable_dns=yes])
+enable_dns_val=0
+if test "x$enable_enable_dns" = "xyes"; then
+    enable_dns_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_DNS], [$enable_dns_val], [])
+AM_CONDITIONAL([ENABLE_DNS], [test x$enable_dns = xyes])
+
+#	CHAOS_DNS_SERVER
```
Comment (ams) [inline configure.ac:56]: @bictorv  Do you think it makes sense to set these values via configure? Or should these be configurable via the config?

Comment (bictorv) [reply configure.ac:56]: What values do you mean? I think it does make sense to be able to un-define e.g. CHAOS_TLS and remove chtls.o and the libs needed for that - and similar for the other "modules".

Comment (ams) [reply configure.ac:56]: That is already possible, `—disable-tls` for example.  What I was referring to was the CHAOS_DNS_SERVER macro, and its other friend that I forgot the name of.

**Comment by ams** [comment]
I did something stupid.

File: configure.ac (hunk 1 above)
Comment (bictorv) [reply configure.ac]: Ah. No, I think the server address (Psilo) should require manual work to change, and also the address domain CH-ADDR.NET.

**Pull request closed without merging**